### PR TITLE
Fix creation of Organization repos by Users with max created personal repos

### LINKED ISSUE #9269: Reached MAX_CREATION_LIMIT forbids user from creating a repo within an org
- Gitea version (or commit ref): 1.9.5
- Operating system: Debian 9
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant

## Description
A user which has reached MAX_CREATION_LIMIT cannot create a repository in an organization where (s)he is an owner.

However when user's MAX_CREATION_LIMIT isn't reached, (s)he can create any number of repositories in an organization. These repositories aren't counted toward the user's MAX_CREATION_LIMIT, which I consider correct. I believe there should be no restriction based on user's MAX_CREATION_LIMIT state for organization owned repos.

...

### PR DESCRIPTION
**Pull request by AndrewBezold**
Creation of Organization repositories by Users with the maximum number of personal repositories is currently broken.  Instead of crashing or giving an error message, don't display Organizations with the maximum number of repositories in the dropdown.

This will not change the initial selection in the dropdown as that is generated from the context.  This will also not remove the user from the list if their number of repositories is at or beyond the maximum.  It will also not affect Administrators, though that is intended.

Fix go-gitea#9269

### CODE REVIEW

File: templates/repo/create.tmpl [hunk 1]
```diff
@@ -31,6 +31,7 @@
 								{{end}}
 							</div>
 						</div>
+						<span class="help">Some organizations may not show up in the dropdown due to the maximum repository limit</span>
```
Comment (guillep2k) [inline templates/repo/create.tmpl]:
```suggestion
						<span class="help">Some organizations may not show up in the dropdown due to a maximum repository count limit</span>
```

File: routers/repo/repo.go [hunk 2]
```diff
@@ -142,7 +147,11 @@ func Create(ctx *context.Context) {
 		}
 	}
 
-	ctx.HTML(200, tplCreate)
+	if !ctx.User.CanCreateRepo() {
```
Comment (guillep2k) [inline routers/repo/repo.go:150]: Actually, `CanCreateRepo` is meant to check more things than it's actually checking. This `if` here ties the failure result to this specific error. Could you please add a note to `CanCreateRepo` like _"NOTE: functions calling this asume a failure due to repository count limit; if new checks are added, those functions should be revised"_?
Just so you don't need to create a specific error object and check for it, which is currently unnecessary.

**Review by guillep2k — APPROVED** [review]

**Review by gary-kim — APPROVED** [review]
LGTM

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**